src/tracer.c

```diff
@@ -1608,13 +1624,13 @@ void __tracer_dormant_to_active(struct snap_device *dev,
         set_bit(ACTIVE, &dev->sd_state);
         clear_bit(UNVERIFIED, &dev->sd_state);
 
-        kfree(cow_path);
+        kfree(resident_cow_path);
```

Comment:
Can this be freed immediately after __tracer_setup_cow_reopen() and then the extra test/free in the error: case could be removed?